File: practical-bdd/java/pages/MyAccountPage.java
package pages;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;

public class MyAccountPage extends BasePage {

    @FindBy(css = "input#username")
    private WebElement emailTextfield;

    @FindBy(css = "input#password")
    private WebElement passwordTextfield;

    @FindBy(css = "[name='login']")
    private WebElement loginButton;

    @FindBy(id = "reg_username")
    private WebElement registrationUsername;

    @FindBy(id = "reg_email")
    private WebElement newUserMail;

    @FindBy(id = "reg_password")
    private WebElement newPassword;

    @FindBy(css = "button[name='register']")
    private WebElement createNewUserButton;

    @FindBy(css = "div.woocommerce-notices-wrapper ul.woocommerce-error")
    private WebElement errorMessage;

    @FindBy(css = "div.woocommerce-MyAccount-content")
    private WebElement myAccountContent;

    public MyAccountPage(WebDriver driver) {
        super(driver);
    }

    public boolean isErrorMessageDisplayed(){
        return errorMessage.isDisplayed();
    }

    public boolean isMyAccountContentDisplayed(){
        return myAccountContent.isDisplayed();
    }

    public boolean isUsernameDisplayed(String username) {
        return myAccountContent.getText().contains(username);
    }

    public void login(String email, String password) {
        emailTextfield.sendKeys(email);
        passwordTextfield.sendKeys(password);
        loginButton.click();
    }

    public void registerNewUser(String username, String emailAddress, String password){
        registrationUsername.sendKeys(username);
        newUserMail.sendKeys(emailAddress);
        newPassword.sendKeys(password);
        createNewUserButton.click();
    }

}

File: practical-bdd/typescript/pages/my-account.page.ts
import { Page } from '@playwright/test';
import { BasePage } from './base.page';

export class MyAccountPage extends BasePage {
  private readonly emailField = this.page.locator('input#username');
  private readonly passwordField = this.page.locator('input#password');
  private readonly loginButton = this.page.locator('[name="login"]');
  private readonly errorMessage = this.page.locator('div.woocommerce-notices-wrapper ul.woocommerce-error');
  private readonly myAccountContent = this.page.locator('div.woocommerce-MyAccount-content');

  constructor(page: Page) {
    super(page);
  }

  async login(email: string, password: string): Promise<void> {
    await this.emailField.fill(email);
    await this.passwordField.fill(password);
    await this.loginButton.click();
  }

  async isErrorMessageDisplayed(): Promise<boolean> {
    return this.errorMessage.isVisible();
  }

  async isMyAccountContentDisplayed(): Promise<boolean> {
    return this.myAccountContent.isVisible();
  }

  async isUsernameDisplayed(username: string): Promise<boolean> {
    const text = await this.myAccountContent.textContent();
    return text?.includes(username) ?? false;
  }
}
